### PR TITLE
Allow using legacy Hive view translation logic

### PR DESCRIPTION
**Pull request by losipiuk**

### CODE REVIEW

File: presto-product-tests/src/main/java/io/prestosql/tests/hive/BaseTestHiveViews.java [hunk 1]
```diff
@@ -38,7 +38,7 @@
         ImmutableNationTable.class,
         ImmutableOrdersTable.class,
 })
-public class TestHiveViews
+public abstract class BaseTestHiveViews
```
Comment (losipiuk) [inline presto-product-tests/src/main/java/io/prestosql/tests/hive/BaseTestHiveViews.java:41]: I am not super convinced that we need this base class.
Maybe just leave `TestHiveViews` as is, and use it for testing default configuration (with Coral).
And add `TestHiveViewsLegacy` based on how `TestHiveViews` looked before Coral was introduced.

File: presto-hive/src/main/java/io/prestosql/plugin/hive/ViewReaderUtil.java [hunk 2]
```diff
@@ -65,6 +68,10 @@ public static ViewReader createViewReader(SemiTransactionalHiveMetastore metasto
         if (isPrestoView(table)) {
             return new PrestoViewReader();
         }
+        if (HiveSessionProperties.isLegacyHiveViewTranslation(session)) {
```
Comment (findepi) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/ViewReaderUtil.java:68]: static import

File: presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveViewsLegacy.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    @BeforeTestWithContext
+    public void setup()
+    {
+        // We need to setup sessions for both "presto" and "default" executors in tempto
+        onPresto().executeQuery("SET SESSION hive.legacy_hive_view_translation = true");
+        query("SET SESSION hive.legacy_hive_view_translation = true");
+    }
+
+    @Test(groups = HIVE_VIEWS)
+    public void testShowCreateView()
```
Comment (findepi) [inline presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveViewsLegacy.java:41]: Did you mean to inherit these tests from the base class?

**Review by findepi — APPROVED** [review]

File: presto-product-tests/src/main/java/io/prestosql/tests/hive/BaseTestHiveViews.java (hunk 1 above)
Comment (findepi) [reply presto-product-tests/src/main/java/io/prestosql/tests/hive/BaseTestHiveViews.java:41]: That would work too.

File: presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveViewsLegacy.java (hunk 3 above)
Comment (losipiuk) [reply presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveViewsLegacy.java:41]: Those are slightly differnt actually.
On the second though I decided to just have separate test classes and drop common base.

**Pull request merged**